lib/mix/lib/mix/tasks/clean.ex

```diff
@@ -36,9 +36,16 @@ defmodule Mix.Tasks.Clean do
       |> Path.join("*#{opts[:only]}")
 
     if opts[:deps] do
-      build
-      |> Path.wildcard()
-      |> Enum.each(&File.rm_rf/1)
+      deps_paths =
+        Mix.Project.deps_paths()
+        |> Map.values()
+        |> Enum.map(&Path.join(&1, "**/*.erl"))
```

Comment:
Unfortunately we cannot simply remove all .erl files because we will remove source files for many erlang projects, that have `.erl` files in their deps. The implementation of this functionality needs to be tied to the `Mix.Task.Compiler` behaviour and it is likely quite more complex than this. :(

Reply:
An alternative could be to modify the compilers that output the `.erl` files either not to do that at all, our output them inside `_build`.

Reply:
What about removing `.erl` files that have their uncompiled version?

If the only way that you saw is going through the `Mix.Task.Compiler` I can try to do this.

Reply:
@michalmuskala I tried that back then but unfortunately I ran into major issues, I don't recall them right now. Modifying the compilers to not output the .erl files actually sounds great. It solves the problem altogether.

Reply:
Yeah, it will also be a fair bit of work, unfortunately. There are some limitations of the yecc and leex compilers themselves to work around to do this